### ISSUE TITLE
Not working with BOA and Cigna

### ISSUE THREAD

**Issue by tmm** [squatto/alfred-imessage-2fa#7]
Just downloaded the workflow from Packal and tried to use with Bank of America and Cigna, but they aren't showing up in Alfred:

<img width="605" alt="image" src="https://user-images.githubusercontent.com/6759464/79481957-2812a800-7fde-11ea-839f-b4be3a8e90d9.png">

Here are the 2fa messages I received from them:

```
<#>BofA: Your code is XXXXXX. Don't share it; we won't call to ask for it. Call 800.933.6262 if you didn't request it.
```

```
Your Cigna authentication code is XXXXXX.
```

**Comment by squatto** [comment]
Thank you for the feedback! Are the codes 6 digits or are there non-digits in the code?

**Comment by tmm** [comment]
They are six digits. No letters or other characters.

**Comment by squatto** [comment]
Every so often I have an issue where macOS doesn't seem to flush the recent messages to the database file. I'm wondering if that is what is happening, especially considering the fact that it *is* reading your other messages. If you reboot and trigger another code being sent to you from Cigna/BoA, does it read the new codes?

**Comment by tmm** [comment]
Just rebooted and tried again with BOA, and unfortunately did not work. Is there another way to update the database file

**Comment by tmm** [comment]
It worked today for a Microsoft account. The only difference I can tell between the Microsoft SMS and Cigna/BOA is that the number they were sent from has a different number of digits.

The Microsoft number had six digits in the following format: XXX-XXX, and the Cigna/BOA had five: XXX-XX.

This thing is magical when it works!

**Comment by manonstreet** [comment]
I was having a similar problem with a number of providers.  The issue is in line 88 of the find-messages.php
`if (preg_match('/(^|\s|\R|\t|G-)(\d{5,8})($|\s|\R|\t)/', $message['text'], $matches)) {`

This code won't detect a code if it is immediately proceeded by a period.  Changing the code to:
`if (preg_match('/(^|\s|\R|\t|G-)(\d{5,8})($|\s|\R|\t|\.)/', $message['text'], $matches)) {`

fixes the issue :)

**Comment by tmm** [comment]
@manonstreet you seem to know what you are doing - do you mind opening up a pull request?

**Comment by squatto** [comment]
This is fixed in [v1.2.1](https://github.com/squatto/alfred-imessage-2fa/releases/tag/v1.2.1) ([download](https://github.com/squatto/alfred-imessage-2fa/releases/download/v1.2.1/iMessage.2FA.alfredworkflow)) - thanks @tmm and @manonstreet!
